lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -115,7 +115,7 @@ protected function createRequest($httpMethod, $path, $body = null, array $header
     public function getLastResponse($force = false)
     {
         $lastResponse =  parent::getLastResponse();
-        if (304 != $lastResponse->getStatusCode()) {
+        if ($lastResponse && 304 != $lastResponse->getStatusCode()) {
```

Comment:
However, I agree with this change. It should be made in a separate PR since it is not related to the other changes.